ui-components/autocompleter.md

```diff
@@ -26,14 +26,14 @@ The two arguments this method take are:
 | Name | Value | Requirement | Description |
 | ---- | ----- | ----------- | ----------- |
 | ch | string | Required | The character to trigger the autocompleter. |
-| fetch |  (pattern: string, maxResults: number) => Promise<AutocompleterItem[]> | Required | A function that is passed the current matched text pattern and the maximum number of expected results. The function should return a promise containing matching results. |
+| fetch |  (pattern: string, maxResults: number, fetchOptions: Record<string, any>) => Promise<AutocompleterItem[]> | Required | A function that is passed the current matched text pattern, the maximum number of expected results and any additional fetch options. The function should return a promise containing matching results. |
```

Comment:
Reading the following paragraphs, is fetchOptions optional? Or are users meant to always pass something, even if it's an empty object? I'm not quite clear on that.

We also don't have a standard way of marking optional args from memory though so not sure how we'd even do that...

Reply:
Actually, I'm also confused on what/where/why fetchOptions is used. Should we explain that more? Is it just meant to be any data that is needed by the fetch implementation to return the right data?

Reply:
Users never call `fetch` it's a function you define that gets called by the system.

Reply:
It's also described in more detail below, but if you have any better suggestions by all means feel free to make a change.